tests/small_numbers/float32_lib.ml

```diff
@@ -0,0 +1,375 @@
+
+(* Tests for the float32 otherlib  *)
```

Comment:
Yeah, a few of these (frexp, ldexp, modf) are exactly the same as the runtime versions, and this is just checking that they're equivalent. I'll add some tests of specific values as well.

The others are different, either because the OCaml versions are primitives/intrinsics or because native compilation uses the libc symbols directly.

Reply:
`compare` also looks like a copy and could use a test of concrete values.

Also this is missing tests for:
* `of_string_opt` (both it and `of_string` should probably be tested on some invalid strings)
* `format`

Reply:
Added more tests for the string operations